CoffeeChat/Controller/EditingViewController.swift

```diff
@@ -257,39 +287,41 @@ extension EditingViewController: UITableViewDelegate {
 // MARK: - UITableViewDataSource
 extension EditingViewController: UITableViewDataSource {
 
-    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        let section = sections[section]
-        switch  section.type {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection sectionIndex: Int) -> Int {
+        switch sections[sectionIndex].type {
         case .yours:
-            return showingLess && yourCount > hideAfter ? hideAfter : yourCount
+            if showingLessRows && yourSectionSize > numberHiddenShownRows {
+                return numberHiddenShownRows
+            } else {
+                return yourSectionSize
+            }
+
         case .more:
-            return moreCount
-      }
+            return moreSectionSize
+        }
     }
 
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let section = sections[indexPath.section]
         let itemType = section.filteredItems[indexPath.row]
 
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: OnboardingTableViewCell.reuseIdentifier, for: indexPath) as? OnboardingTableViewCell else { return UITableViewCell() }
+
+        cell.changeColor(isSelected: section.type == .yours)
```

Comment:
Any reason you do this before the switch? It would make more sense to change the color after, so all cell state/property changes occur in the same place.